frontend/src/lib/components/SigninWithGoogleButton.svelte

```diff
@@ -0,0 +1,172 @@
+<script lang="ts">
+  export let href: string;
+  import t from '$lib/i18n';
+</script>
+
+<div class="google-signin-top-level" style="position: relative;">
```

Comment:
When you use the Google Javascript library, and you're *not* signed in, it creates a "Log in with Google" button in your page. I copied its HTML and CSS, then transcribed them into this component. So that way I could ensure that it's complying with their https://developers.google.com/identity/branding-guidelines, because it's exactly the same HTML and CSS (with different class names, but the same page layout) as what their own JS library generates.